docs/contracts.rst

```diff
@@ -990,12 +1020,28 @@ virtual method lookup.
 
 .. index:: ! constructor
 
+.. _constructor:
+
 Constructors
 ============
-A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation.
-Constructor functions can be either ``public`` or ``internal``. If there is no constructor, the contract will assume the
-default constructor: ``contructor() public {}``.
 
+A constructor is an optional function declared with the ``constructor`` keyword
+which is executed upon contract creation, and where you can run contract
+initialisation code.
+
+Other code executed once right before the constructor always generates code to
+represent the initial storage state, initialising any state variables you
+declare to their specified default value if you initialise them inline, or zero
+if you do not.
+
+After that it returns the final code of the contract. The final deployment of
+the code costs additional gas linear to the length of the code. If you did not
+supply enough gas to initiate the state variables declared in the constructor,
+then an "out of gas" exception is generated. This is the **only** situation
```

Comment:
Is this still true? I don't think so.

Reply:
@chriseth is what true?

> If you did not supply enough gas to initiate the state variables declared in the constructor, then an "out of gas" exception is generated. 

Or

> This is the **only** situation where an "out of gas" exception does not revert changes to the state.

If that's the case, then you know best and I'll change :)

Reply:
only

Reply:
I'm quite sure this can be removed.